include/nlohmann/detail/conversions/from_json.hpp

```diff
@@ -152,8 +153,19 @@ void from_json(const BasicJsonType& j, std::valarray<T>& l)
     {
         JSON_THROW(type_error::create(302, "type must be array, but is " + std::string(j.type_name())));
     }
-    l.resize(j.size());
-    std::copy(j.m_value.array->begin(), j.m_value.array->end(), std::begin(l));
+    std::valarray<T> ret(j.size());
+    std::copy(j.m_value.array->begin(), j.m_value.array->end(), std::begin(ret));
```

Comment:
Why is this change necessary?  It forces a new allocation where before it could reuse existing storage?

Reply:
True